project/Build.scala

```diff
@@ -817,6 +817,74 @@ object Build {
+    settings(commonBootstrappedSettings).
+    settings(
+      javaOptions := (javaOptions in `dotty-compiler-bootstrapped`).value,
+      javaOptions += "-Ddotty.scala.library=" + packageBin.in(`dotty-scala-library`, Compile).value.getAbsolutePath
+    )
+
+
+  // Scala library compiled by dotty using the latest published sources of the library
+  lazy val `dotty-scala-library` = project.in(file("dotty-scala-library")).
+    withCommonSettings(Bootstrapped).
+    dependsOn(dottyCompiler(Bootstrapped) % "provided; compile->runtime; test->test").
+    dependsOn(`dotty-tasty-inspector` % "test->test").
+    settings(commonBootstrappedSettings).
+    settings(
+      javaOptions := (javaOptions in `dotty-compiler-bootstrapped`).value,
+      scalacOptions -= "-Xfatal-warnings",
+      ivyConfigurations += SourceDeps.hide,
+      transitiveClassifiers := Seq("sources"),
+      libraryDependencies +=
+        ("org.scala-lang" % "scala-library" % stdlibVersion(Bootstrapped) % "sourcedeps"),
+      sourceGenerators in Compile += Def.task {
+        val s = streams.value
+        val cacheDir = s.cacheDirectory
+        val trgDir = (sourceManaged in Compile).value / "scala-library-src"
+
+        val report = updateClassifiers.value
+        val scalaLibrarySourcesJar = report.select(
+            configuration = configurationFilter("sourcedeps"),
+            module = (_: ModuleID).name == "scala-library",
+            artifact = artifactFilter(`type` = "src")).headOption.getOrElse {
+          sys.error(s"Could not fetch scala-library sources")
+        }
+
+        FileFunction.cached(cacheDir / s"fetchScalaLibrarySrc",
+            FilesInfo.lastModified, FilesInfo.exists) { dependencies =>
+          s.log.info(s"Unpacking scala-library sources to $trgDir...")
+          if (trgDir.exists)
+            IO.delete(trgDir)
+          IO.createDirectory(trgDir)
+          IO.unzip(scalaLibrarySourcesJar, trgDir)
+
+          ((trgDir ** "*.scala") +++ (trgDir ** "*.java")).get.toSet
+        } (Set(scalaLibrarySourcesJar)).toSeq
+      }.taskValue,
+      sources in Compile ~= (_.filterNot(file =>
+        // sources from https://github.com/scala/scala/tree/2.13.x/src/library-aux
+        file.getPath.endsWith("scala/Any.scala") ||
+        file.getPath.endsWith("scala/AnyVal.scala") ||
+        file.getPath.endsWith("scala/AnyRef.scala") ||
+        file.getPath.endsWith("scala/Nothing.scala") ||
+        file.getPath.endsWith("scala/Null.scala") ||
+        file.getPath.endsWith("scala/Singleton.scala")))
+    )
+
+  // Tests that we can compile and run an app using the scala-library compiled by dotty
+  lazy val `dotty-scala-library-sandbox` = project.in(file("sandbox/dotty-scala-library")).
+    withCommonSettings(Bootstrapped).
+    dependsOn(`dotty-scala-library`).
+    settings(
+      managedClasspath in Compile ~= {
+        _.filterNot(file => file.data.getName == "scala-library-2.13.3.jar")
```

Comment:
If you want to remove the stdlib from the dependencies of a scala project, you can use `autoScalaLibrary := false` (this will also remove dotty-library)

Reply:
We don't want to remove the dotty-library.

Reply:
Ok, but then don't hardcore 2.13.3, use `stdlibVersion`